checks.d/haproxy.py

```diff
@@ -287,9 +295,21 @@ def _tag_match_patterns(self, tag, filters):
                 return True
         return False
 
+    @staticmethod
+    def _normalize_status(status):
+        """
+        Try to match the HAProxy status with one of the statuses defined in `ALL_STATUSES`,
+        if it can't be matched return the status as-is in a tag-friendly format
+        """
+        formatted_status = status.lower().replace(" ", "_")
```

Comment:
replacing spaces with underscores is needed when the raw status is `no check` for instance, but we could do it after we've tried matching the status with the "known" statuses